examples/isometric/game.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2021 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build example
+// +build example
+
+package main
+
+import (
+	"fmt"
+	"image"
+
+	"github.com/hajimehoshi/ebiten/v2"
+	"github.com/hajimehoshi/ebiten/v2/ebitenutil"
+	"github.com/hajimehoshi/ebiten/v2/inpututil"
+)
+
+var spinner = []byte(`-\|/`)
+
+// Game is an isometric demo game.
+type Game struct {
+	w, h         int
+	currentLevel *Level
+
+	camX, camY float64
+	camScale   float64
+	camScaleTo float64
+
+	mousePanX, mousePanY int
+
+	spinnerIndex int
+}
+
+// NewGame returns a new isometric demo Game.
+func NewGame() *Game {
+	return &Game{
+		currentLevel: NewLevel(),
+		camScale:     2,
+		camScaleTo:   2,
+	}
+}
+
+// Update reads current user input and updates the Game state.
+func (g *Game) Update() error {
+	// Update target zoom level.
+	var scrollY float64
+	if ebiten.IsKeyPressed(ebiten.KeyC) || ebiten.IsKeyPressed(ebiten.KeyPageDown) {
+		scrollY = -0.15
+	} else if ebiten.IsKeyPressed(ebiten.KeyE) || ebiten.IsKeyPressed(ebiten.KeyPageUp) {
+		scrollY = .15
+	} else {
+		_, scrollY = ebiten.Wheel()
+		if scrollY < -1 {
+			scrollY = -1
+		} else if scrollY > 1 {
+			scrollY = 1
+		}
+	}
+	g.camScaleTo += scrollY * (g.camScaleTo / 7)
+
+	// Clamp target zoom level.
+	if g.camScaleTo < 0.01 {
+		g.camScaleTo = 0.01
+	} else if g.camScaleTo > 100 {
+		g.camScaleTo = 100
+	}
+
+	// Smooth zoom transition.
+	div := 10.0
+	if g.camScaleTo > g.camScale {
+		g.camScale += (g.camScaleTo - g.camScale) / div
+	} else if g.camScaleTo < g.camScale {
+		g.camScale -= (g.camScale - g.camScaleTo) / div
+	}
+
+	pan := 4.0 / g.camScale
+	// Pan camera via keyboard.
+	if ebiten.IsKeyPressed(ebiten.KeyLeft) || ebiten.IsKeyPressed(ebiten.KeyA) {
+		g.camX -= pan
+	}
+	if ebiten.IsKeyPressed(ebiten.KeyRight) || ebiten.IsKeyPressed(ebiten.KeyD) {
+		g.camX += pan
+	}
+	if ebiten.IsKeyPressed(ebiten.KeyDown) || ebiten.IsKeyPressed(ebiten.KeyS) {
+		g.camY -= pan
+	}
+	if ebiten.IsKeyPressed(ebiten.KeyUp) || ebiten.IsKeyPressed(ebiten.KeyW) {
+		g.camY += pan
+	}
+	// Pan camera via mouse.
+	if ebiten.IsMouseButtonPressed(ebiten.MouseButtonRight) {
+		if g.mousePanX == 0 && g.mousePanY == 0 {
```

Comment:
0 seems a valid value as CursorPosition. What about using `math.MinInt32` indicating an invalid value?

Reply:
Or

```
	maxInt     = int(^uint(0) >> 1)
	minInt     = -maxInt - 1
```

and use `minInt`